src/services/factories/tokenList.ts

```diff
@@ -130,42 +130,57 @@ export function getTokensFactory(factoryParams: {
     return new Set(tcrApi && (await tcrApi.getTokens(networkId)))
   }
 
-  async function updateTokenDetails(networkId: number, numTokens: number): Promise<void> {
-    const [tcrAddressesSet, addressesAndIds] = await Promise.all([
+  async function _getFilteredIdsMap(networkId: number, numTokens: number): Promise<Map<string, number>> {
+    const [tcrAddressesSet, listedAddressesAndIds] = await Promise.all([
       // Fetch addresses from TCR
       retry(() => fetchTcrAddresses(networkId)),
       // Fetch addresses from contract given numTokens count
       retry(() => fetchAddressesAndIds(networkId, numTokens)),
     ])
 
-    logDebug(`[tokenListFactory][${networkId}] TCR contains ${tcrAddressesSet.size} addresses`)
+    if (tcrAddressesSet.size > 0) {
+      // If filtering by TCR
+      logDebug(`[tokenListFactory][${networkId}] TCR contains ${tcrAddressesSet.size} addresses`)
+      logDebug(`[tokenListFactory][${networkId}] Token id and address mapping:`)
+      const filteredAddressAndIds = Array.from(listedAddressesAndIds.entries()).filter(([address, id]) => {
+        // Remove addresses that are not on the TCR, if any
+        const isInTcr = tcrAddressesSet.has(address)
+        logDebug(`[tokenListFactory][${networkId}] ${id} : ${address}. On TCR? ${isInTcr}`)
+      })
+      return new Map<string, number>(filteredAddressAndIds)
+    } else {
+      // If not using a TCR, filter by default list
+      logDebug(`[tokenListFactory][${networkId}] Not using a TCR. Filtering by tokens in the config file`)
+      const tokensConfig = tokenListApi.getTokens(networkId)
+      const filteredAddressAndIds = tokensConfig
+        .map((token): [string, number] | null => {
+          const tokenId = listedAddressesAndIds[token.address]
+          return [token.address, tokenId] || null
+        })
+        .filter(notEmpty)
+
+      return new Map<string, number>(filteredAddressAndIds)
```

Comment:
you don't want to move this outside the `if/else` with a `let filteredAddressAndIds` outside the scope? seeing as you're returning `new Map(filteredAddressAndIds` in both cases

Reply:
I considered it, but just went for this option. I'm ok change it too. In this case I don't feel makes a huge difference

Reply:
5b8e19c3e83925d232bdfb2ac7dcbab98bb0b40f